src/scaladoc/scala/tools/nsc/doc/html/Doclet.scala

```diff
@@ -7,14 +7,19 @@ package scala.tools.nsc
 package doc
 package html
 
+import scala.reflect.internal.Reporter
 import doclet._
 
 /** The default doclet used by the scaladoc command line tool
   * when no user-provided doclet is provided. */
-class Doclet extends Generator with Universer {
+class Doclet(reporter: Reporter) extends Generator with Universer {
 
-  def generateImpl() {
-    new html.HtmlFactory(universe, new ScalaDocReporter(universe.settings)).generate()
-  }
+  @deprecated("Doclets should be created with the Reporter constructor. Otherwise logging reporters will not be shared by the creating parent", "2.12.0")
+  def this() { this(null) }
```

Comment:
I could change it to `def this() = this(null)`, however, I think it'd be hard to change it to:

`def this() = this(new ScalaDocReporter)` since the reporter constructor takes a settings value and that setting is created by the `universe` variable, which in turn isn't set until somebody calls `setUniverse` :(